.github/workflows/deploy-happy-stack.yml

```diff
@@ -162,6 +162,140 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  run-e2e-tests:
```

Comment:
similar sharded tests as the `push-test` one. It's not completely identical due to the fact that this workflow needs to be aware of different deployed environments and slightly different status reporting mechanism as a result of that